From 5303b302939a854ddd4e712046a20fa502b51ef9 Mon Sep 17 00:00:00 2001
From: jakub <jakub@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Tue, 9 Apr 2019 06:38:07 +0000
Subject: [PATCH 34/42] 	PR target/90015 	* config/riscv/riscv.c
 (riscv_get_interrupt_type): Fix comment typo. 	(riscv_merge_decl_attributes):
 Fix typo in diagnostics.  Remove 	trailing period from it too.

	* gcc.target/riscv/interrupt-conflict-mode.c (foo): Adjust expected
	diagnostics.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@270221 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog                                            | 8 ++++++++
 gcc/config/riscv/riscv.c                                 | 4 ++--
 gcc/testsuite/ChangeLog                                  | 6 ++++++
 gcc/testsuite/gcc.target/riscv/interrupt-conflict-mode.c | 2 +-
 4 files changed, 17 insertions(+), 3 deletions(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index fac19cfdbb5..0960169381c 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,11 @@
+2019-04-09  Jakub Jelinek  <jakub@redhat.com>
+
+	PR target/90015
+	* config/riscv/riscv.c (riscv_get_interrupt_type): Fix comment typo.
+	(riscv_merge_decl_attributes): Fix typo in diagnostics.  Remove
+	trailing period from it too.
+
+
 2019-07-08  Andrew Waterman  <andrew@sifive.com>
 	    Jim Wilson  <jimw@sifive.com>
 
diff --git a/gcc/config/riscv/riscv.c b/gcc/config/riscv/riscv.c
index e488a6482d2..0b6069bfcd6 100644
--- a/gcc/config/riscv/riscv.c
+++ b/gcc/config/riscv/riscv.c
@@ -4768,7 +4768,7 @@ riscv_function_ok_for_sibcall (tree decl ATTRIBUTE_UNUSED,
   return true;
 }
 
-/* Get the intterupt type, return UNKNOWN_MODE if it's not
+/* Get the interrupt type, return UNKNOWN_MODE if it's not
    interrupt function. */
 static enum riscv_privilege_levels
 riscv_get_interrupt_type (tree decl)
@@ -4854,7 +4854,7 @@ riscv_merge_decl_attributes (tree olddecl, tree newdecl)
   if ((old_interrupt_type != UNKNOWN_MODE)
       && (new_interrupt_type != UNKNOWN_MODE)
       && (old_interrupt_type != new_interrupt_type))
-    error ("%qs function cannot have different intterupt type.", "interrupt");
+    error ("%qs function cannot have different interrupt type", "interrupt");
 
   /* Create combined attributes.  */
   combined_attrs = merge_attributes (DECL_ATTRIBUTES (olddecl),
diff --git a/gcc/testsuite/ChangeLog b/gcc/testsuite/ChangeLog
index 097677ae340..58127bed0e0 100644
--- a/gcc/testsuite/ChangeLog
+++ b/gcc/testsuite/ChangeLog
@@ -1,3 +1,9 @@
+2019-04-09  Jakub Jelinek  <jakub@redhat.com>
+
+	PR target/90015
+	* gcc.target/riscv/interrupt-conflict-mode.c (foo): Adjust expected
+	diagnostics.
+
 2019-07-08  Jim Wilson  <jimw@sifive.com>
 
 	* gcc.target/riscv/shift-shift-2.c: Add one more test.
diff --git a/gcc/testsuite/gcc.target/riscv/interrupt-conflict-mode.c b/gcc/testsuite/gcc.target/riscv/interrupt-conflict-mode.c
index 6e280ed1771..e9f145265c0 100644
--- a/gcc/testsuite/gcc.target/riscv/interrupt-conflict-mode.c
+++ b/gcc/testsuite/gcc.target/riscv/interrupt-conflict-mode.c
@@ -6,5 +6,5 @@ foo(void);
 
 void __attribute__ ((interrupt ("machine")))
 foo (void)
-{ /* { dg-error "function cannot have different intterupt type." } */
+{ /* { dg-error "function cannot have different interrupt type" } */
 }
-- 
2.20.1

